test/iterators.jl

```diff
@@ -133,6 +133,12 @@ let i = 0, k = 1
         k += 1
         k <= 10 || break
     end
+    for j = countfrom(Int[0,0], Float64[1.0,2.0])
+        @test j isa Vector{Float64}
+        @test j == l*[1,2]
+        l += 1
+        l <= 10 || break
+    end
```

Comment:
Maybe have a test case for types where T and S are different, e.g.
```jl
@test collect(take(countfrom(Dates.Date("2020-12-25"), Dates.Day(1)), 12)) == range(Dates.Date("2020-12-25"), step=Dates.Day(1), length=12)
```

Reply:
indeed, here T is promoted to `Vector{Float64}` as well
is it ok to use `Dates` in tests or should I use types from Base Julia ?